staging/src/k8s.io/apiserver/pkg/audit/request.go

```diff
@@ -52,6 +52,8 @@ func LogRequestMetadata(ctx context.Context, req *http.Request, requestReceivedT
 	ev.RequestURI = req.URL.RequestURI()
 	ev.UserAgent = maybeTruncateUserAgent(req)
 	ev.Level = level
+	ev.TLSExpiration = metav1.NewMicroTime(req.TLS.PeerCertificates[0].NotAfter)
```

Comment:
this will panic on requests that don't have certificates

Reply:
this needs a little thought... some questions:
1. should this only be set when client cert was the authentication mechanism used?
2. should this be top-level or within an authentication stanza?
3. is credential expiration a generic thing we should surface? (service account tokens and oidc tokens also have associated expirations

Reply:
1. Audit log should record important info includes client cert, even if k8s don't use it, we should let k8s manager know what happened in a request.
2. if we want to add other authentication info like token expiration to audit log, I think an authentication stanza is needed.
3. as I mentioned before, k8s exposed certificate_expiration_seconds metric to remind clients this problem, and in fact there is no way to find out which client is using it; and it sure is a problem that client use expired authentication or close to expire.